src/components/RouteChangeHandler/RouteChangeHandler.js

```diff
@@ -0,0 +1,110 @@
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+import { ELEMENT_IDS, NO_CATEGORY } from '../../constants';
+
+/**
+ * Add an event listener which only listens once
+ * There is a { once: true } parameter on addEventListener,
+ * but it doesn't seem well-supported ATM
+ */
+const addOneTimeEventListener = (el, event, fn) => {
+  const func = () => {
+    el.removeEventListener(event, func);
+    fn();
+  };
+
+  el.addEventListener(event, func);
+};
+
+/**
+ * If focus point is found, set focus without a visual style until user tabs
+ */
+function focus(oldPath, currentPath) {
+  let focusPoint;
+
+  // first look for a hash
+  if (document.location.hash) {
+    focusPoint = document.querySelector(document.location.hash);
+  }
+
+  // if we have changed cities, and landed on overview
+  // then focus on overview subnav item
+  if (!focusPoint) {
+    const oldSegments = oldPath.split('/', 3);
+    const currentSegments = currentPath.split('/', 3);
+
+    if (currentSegments[2] === NO_CATEGORY && currentSegments[1] !== oldSegments[1]) {
+      focusPoint = document.getElementById(ELEMENT_IDS.OVERVIEW_LINK);
+    }
+  }
+
+  // if there was no hash, or hash didn't point to anything,
+  // look for a manually set focus point ID
+  if (!focusPoint) {
+    focusPoint = document.getElementById(ELEMENT_IDS.INITIAL_FOCUS_POINT_ON_PAGE);
+  }
+
+  if (!focusPoint) {
+    return;
+  }
+
+  const outlineStyle = focusPoint.style.outline;
+
+  addOneTimeEventListener(focusPoint, 'blur', () => {
+    focusPoint.style.outline = outlineStyle;
```

Comment:
This needs to `focusPoint.removeAttribute('tabIndex')` otherwise:

![image](https://user-images.githubusercontent.com/4443482/33345658-35e50702-d4e1-11e7-9236-8d072a4d1d45.png)

Reply:
DOH!